src/plugins/kibana_react/public/util/mount_point_portal.tsx

```diff
@@ -0,0 +1,57 @@
+/*
+ * Licensed to Elasticsearch B.V. under one or more contributor
+ * license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright
+ * ownership. Elasticsearch B.V. licenses this file to you under
+ * the Apache License, Version 2.0 (the "License"); you may
+ * not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+import React, { useRef, useEffect, useState } from 'react';
+import ReactDOM from 'react-dom';
+import { MountPoint } from 'kibana/public';
+
+interface MountPointPortalProps {
+  setMountPoint: (mountPoint: MountPoint<HTMLElement>) => void;
+}
+
+/**
+ * Utility component to portal a part of a react application into the provided `MountPoint`.
+ */
+export const MountPointPortal: React.FC<MountPointPortalProps> = ({ children, setMountPoint }) => {
+  // state used to force re-renders when the element changes
+  const [shouldRender, setShouldRender] = useState(false);
+  const el = useRef<HTMLElement>();
+
+  useEffect(() => {
+    setMountPoint((element) => {
+      el.current = element;
+      setShouldRender(true);
+      return () => {
+        setShouldRender(false);
+        el.current = undefined;
+      };
+    });
+
+    return () => {
+      setShouldRender(false);
+      el.current = undefined;
+    };
+  }, [setMountPoint]);
+
+  if (shouldRender && el.current) {
+    return ReactDOM.createPortal(children, el.current);
```

Comment:
Should we have a default error boundary here?

Reply:
Wasn't sure. Added it.